File: dp/kadane-_algorithm/rust/kadane.rs
fn kadane(v: &Vec<i32>) -> i32{
    let mut max_global = v[0];
    let mut max_current = max_global;

    for i in 1..(v.len()-1) {
        max_current = if v[i] > max_current + v[i] {
            v[i]
        } else {
            max_current + v[i]
        };

        if max_current > max_global {
            max_global = max_current;
        }
    }

    return max_global;
}

fn main() {
    let v: Vec<i32> = vec![-2, 3, 2, -1];

    println!("Maximum subarray sum: {}", kadane(&v));
}

File: dp/kadane-_algorithm/cpp/kadane.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
	vector<int> v={-2,-1,-5,3,7,-2,5,11,-10,-20,11};
	int n=v.size();
	int mini=*min_element(v.begin(),v.end());
	int maxval=mini,curval=mini;
	for(int i=0;i<n;i++){
		curval=max(curval+v[i],v[i]);
		maxval=max(maxval,curval);
	}
	cout<<"Maximum Subarray Sum = "<<maxval<<endl;
}